### DISCUSSION TITLE
how to show exception traces?

### DISCUSSION THREAD

**Discussion by scott2b** [litestar-org/litestar#780] [Q&A] [answered]
How do I get the server to show my application exceptions in development? I guess I figured that's what the debug flag is for, but it still just logs out a 500 when debug is enabled.

**Comment by scott2b** [comment, marked as answer]
Ok, I see now that debug sends the stack trace as the payload to the client. So, e.g. with httpx, you can get this from the text property of the response.

And, I guess, if really needed, you can raise the exception on the server with an after_exception handler like:

```python
async def after_exception_handler(exc: Exception, scope: "Scope", state: "State") -> None:
    raise Exception

...

app = Starlite(after_exception=after_exception_handler ....
```

**Reply by Goldziher** [reply]
> Ok, I see now that debug sends the stack trace as the payload to the client. So, e.g. with httpx, you can get this from the text property of the response.
> 
> And, I guess, if really needed, you can raise the exception on the server with an after_exception handler like:
> 
> ```
> async def after_exception_handler(exc: Exception, scope: "Scope", state: "State") -> None:
>     raise Exception
> 
> ...
> 
> app = Starlite(after_exception=after_exception_handler ....
> ```

You can also add an app level exception handler that logs exceptions.

**Reply by scott2b** [reply]
Docs for the logging approach @Goldziher mentions are [here](https://starlite-api.github.io/starlite/usage/17-exceptions/#logging-exception-handler)
